testing/trino-product-tests/src/main/java/io/trino/tests/product/hudi/TestHudiSparkCompatibility.java

```diff
@@ -0,0 +1,325 @@
+            expectedRows = ImmutableList.of(
+                    row(1, "a1_1", 1001),
+                    row(2, "a2", 1000));
+            assertThat(onHudi().executeQuery("SELECT id, name, ts FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT id, name, ts FROM hudi.default." + tableName))
+                    .containsOnly(expectedRows);
+        }
+        finally {
+            onHudi().executeQuery("DROP TABLE default." + tableName);
+        }
+    }
+
+    @Test(groups = {HUDI, PROFILE_SPECIFIC_TESTS})
+    public void testMergeOnReadTableSelect()
+    {
+        String tableName = "test_hudi_mor_select_" + randomTableSuffix();
+        String tableDirectory = "hudi-compatibility-test-" + tableName;
+
+        createNonPartitionedTable(tableName, tableDirectory, MOR_TABLE_TYPE);
+
+        List<QueryAssert.Row> expectedRows = ImmutableList.of(
+                row(1, "a1", 20, 1000),
+                row(2, "a2", 40, 2000));
+
+        try {
+            assertThat(onHudi().executeQuery("SELECT id, name, price, ts FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT id, name, price, ts FROM hudi.default." + tableName))
+                    .containsOnly(expectedRows);
+        }
+        finally {
+            onHudi().executeQuery("DROP TABLE default." + tableName);
+        }
+    }
+
+    @Test(groups = {HUDI, PROFILE_SPECIFIC_TESTS})
+    public void testMergeOnReadTableSelectAfterUpdate()
+    {
+        String tableName = "test_hudi_mor_update" + randomTableSuffix();
+        String tableDirectory = "hudi-compatibility-test-" + tableName;
+
+        createNonPartitionedTable(tableName, tableDirectory, MOR_TABLE_TYPE);
+
+        List<QueryAssert.Row> expectedRows = ImmutableList.of(
+                row(1, "a1", 20, 1000),
+                row(2, "a2", 40, 2000));
+
+        try {
+            assertThat(onHudi().executeQuery("SELECT id, name, price, ts FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT id, name, price, ts FROM hudi.default." + tableName))
+                    .containsOnly(expectedRows);
+
+            onHudi().executeQuery("UPDATE default." + tableName + " SET ts = 2020 WHERE id = 2");
+            List<QueryAssert.Row> expectedRowsAfterUpdate = ImmutableList.of(
+                    row(1, "a1", 20, 1000),
+                    row(2, "a2", 40, 2020));
+            assertThat(onHudi().executeQuery("SELECT id, name, price, ts FROM default." + tableName))
+                    .containsOnly(expectedRowsAfterUpdate);
+            // NOTE: MOR Snapshot queries are not supported yet.
+            // "_ro" suffix to the table indicates read-optimized query.
+            // See https://hudi.apache.org/docs/docker_demo/#step-4-d-run-trino-queries for more info.
```

Comment:
Let's remove the link to the demo page - this may eventually change sooner or later.